src/code.cloudfoundry.org/loggregator/plumbing/tls.go

```diff
@@ -75,40 +52,87 @@ func WithCipherSuites(ciphers []string) ConfigOption {
 	}
 }
 
-// NewMutualTLSConfig returns a tls.Config with certs loaded from files and
+// NewClientMutualTLSConfig returns a tls.Config with certs loaded from files and
 // the ServerName set.
-func NewMutualTLSConfig(
+func NewClientMutualTLSConfig(
+	certFile string,
+	keyFile string,
+	caCertFile string,
+	serverName string,
+) (*tls.Config, error) {
+	return newMutualTLSConfig(
+		certFile,
+		keyFile,
+		caCertFile,
+		serverName,
+	)
+}
+
+// NewServerMutualTLSConfig returns a tls.Config with certs loaded from files and
+// the ServerName set. The returned tls.Config has configured list of cipher
+// suites.
+func NewServerMutualTLSConfig(
 	certFile string,
 	keyFile string,
 	caCertFile string,
 	serverName string,
 	opts ...ConfigOption,
 ) (*tls.Config, error) {
-	tlsCert, err := tls.LoadX509KeyPair(certFile, keyFile)
+	tlsConfig, err := newMutualTLSConfig(
+		certFile,
+		keyFile,
+		caCertFile,
+		serverName,
+	)
 	if err != nil {
-		return nil, fmt.Errorf("failed to load keypair: %s", err.Error())
+		return nil, err
 	}
 
-	tlsConfig := &tls.Config{
-		InsecureSkipVerify: false,
-		MinVersion:         tls.VersionTLS12,
-		CipherSuites:       defaultCipherSuites,
-	}
+	tlsConfig.CipherSuites = defaultCipherSuites
 	for _, opt := range opts {
 		opt(tlsConfig)
 	}
 
-	tlsConfig.Certificates = []tls.Certificate{tlsCert}
-	tlsConfig.ClientAuth = tls.RequireAndVerifyClientCert
-	tlsConfig.ServerName = serverName
+	return tlsConfig, nil
+}
 
-	if caCertFile != "" {
-		if err := addCA(tlsConfig, tlsCert, caCertFile); err != nil {
-			return nil, err
-		}
+func NewClientCredentials(
+	certFile string,
+	keyFile string,
+	caCertFile string,
+	serverName string,
+) (credentials.TransportCredentials, error) {
+	tlsConfig, err := NewClientMutualTLSConfig(
+		certFile,
+		keyFile,
+		caCertFile,
+		serverName,
+	)
+	if err != nil {
+		return nil, err
 	}
 
-	return tlsConfig, err
+	return credentials.NewTLS(tlsConfig), nil
+}
+
+func NewServerCredentials(
+	certFile string,
+	keyFile string,
+	caCertFile string,
+	serverName string,
+	opts ...ConfigOption,
```

Comment:
Unfortunately because the type is wrapped in `credentials.TransportCredentials` which does not expose any information we are setting. Therefore we can't assert against the opts being passed through.